Documentation/Storage-Configuration/Object-Storage-RGW/ceph-object-swift.md

```diff
@@ -0,0 +1,136 @@
+      revocationInterval: 1200
+      serviceUserSecretName: usersecret
+      tokenCacheSize: 1000
+      url: https://keystone.rook-ceph.svc/
+  protocols:
+    swift:
+      accountInUrl: true
+      urlPrefix: /swift  
+  preservePoolsOnDelete: true
+  gateway:
+    sslCertificateRef:
+    port: 80
+    # securePort: 443
+    instances: 1
+```
+
+After the `CephObjectStore` is created, the Rook operator will create all the pools and other resources necessary to start the service. This may take a minute to complete.
+
+```console
+kubectl create -f object.yaml
+```
+
+The start of the RGW pod(s) confirms that the object store is configured.
+
+```console
+kubectl -n rook-ceph get pod -l app=rook-ceph-rgw
+```
+
+The swift service endpoint in OpenStack/Keystone must be created, in order to use the object store in Swift using for example the [OpenStack CLI](https://docs.openstack.org/python-openstackclient/latest/).
+The endpoint url should be set to the service endpoint of the created rgw instance.
+
+```sh
+openstack service create --name swift object-store
+openstack endpoint create --region default --enable swift admin https://rook-ceph-rgw-default.rook-ceph.svc/swift/v1
+openstack endpoint create --region default --enable swift internal https://rook-ceph-rgw-default.rook-ceph.svc/swift/v1
+```
+
+Afterwards any user which has the rights to access the projects resources (as defined in the OpenStack Keystone instance) can access the object store and create container and objects.
+Here the username and project are explicitly set to reflect use of the (non-admin) user.
+
+```sh
+export OS_USERNAME=alice
+export OS_PROJECT=exampleProject
+openstack container create exampleContainer
+# put /etc/hosts in the new created container
+openstack object create exampleContainer /etc/hosts
+# retrieve and save the file
+openstack object save --file /tmp/hosts.saved exampleContainer /etc/hosts
+openstack object delete exampleContainer /etc/hosts
+openstack container delete exampleContainer 
+```
+
+## Basic concepts
+
+When using Keystone as an authentication provider, Ceph uses the credentials of an admin user (provided in the secret references by `serviceUserSecretName`) to access Keystone.
+
+For each user accessing the object store using Swift, Ceph implicitly creates a user which must be represented in Keystone with an authorized counterpart.
+Keystone checks for a user of the same name. Based on the name and other parameters ((OpenStack Keystone) project, (OpenStack Keystone) role) Keystone allows or disallows access to a swift container or object. 
+
+It is not necessary to create any users in OpenStack Keystone (except for the admin user provided in the `serviceUserSecretName`).
+
+## Keystone setup
+
+Keystone must support the v3-API-Version to be used with Rook . Other API versions are not supported.
+
+The admin user and all users accessing the Object store must exist and their authorizations configured accordingly in Keystone.
```

Comment:
Swift has always been enabled by default and this PR doesn't change that. It only adds a section to configure Swift. So if that section is omitted, Swift might just not work properly. If S3 is enabled, it will work with Keystone.

If Swift is configured without Keystone, it uses its default authentication mechanism (as it is doing already) which means that users have to be manually created via `radosgw-admin`. Unauthenticated access is never possible.